### PR TITLE
cmd: add `alpha view-cluster-manifest` command

### LINKED ISSUE #2381: Add command to view cluster manifests
# 🎯 Problem to be solved

The cluster manifest files are in binary (protobuf) format so is not human readable. But often times, we want to check the contents of a cluster manifest. For example, we may want to check if the manifest contains our new validators after running the `add-validators-solo` command.

We should provide a way for our users to easily check contents of a cluster manifest file.

# 🛠️ Proposed solution

Add a new command `charon alpha view-cluster-manifest` which:
* loads a manifest file from path specified by the `--manifest-file` flag
* reads the cluster data from the file
* prints the cluster data in a human-readable format (json)

Note that we can probably use the [protojson package](https://pkg.go.dev/google.golang.org/protobuf/encoding/protojson) for this.

# 🧪 Tests
- [ ] Tested by new automated unit/integration/smoke tests
- [ ] Manually tested on core team/canary/test clusters
- [ ] Manually tested on local compose simnet

### PR DESCRIPTION
**Pull request by gsora**
Moved some manifest load/store functions to `manifest_tools.go` and generalized them, so we can reuse code across multiple commands.

Closes #2381 

category: feature
ticket: #2381

### CODE REVIEW

**Review by corverroos — APPROVED** [review]

**Review by corverroos — APPROVED** [review]

File: cmd/addvalidators.go [hunk 1]
```diff
@@ -93,11 +92,23 @@ func bindAddValidatorsFlags(cmd *cobra.Command, config *addValidatorsConfig) {
...
+		if err != nil {
+			return err
+		}
+	} else {
+		// Read lock file to load cluster manifest
+		cluster, _, err = loadClusterManifest(conf.ManifestFile, conf.Lockfile)
+		if err != nil {
+			return err
+		}
 	}
```
Comment (xenowits) [inline cmd/addvalidators.go:110]: nit: we can probably move this logic to the `loadClusterManifest()` function. The function then should return a cluster manifest either from TestConfig or by loading from disk.

File: cmd/cmd.go [hunk 2]
```diff
@@ -46,6 +46,7 @@ func New() *cobra.Command {
 		newCombineCmd(newCombineFunc),
 		newAlphaCmd(
 			newAddValidatorsCmd(runAddValidatorsSolo),
+			newViewClusterManifestCmd(runViewClusterManifest),
```
Comment (xenowits) [inline cmd/cmd.go:49]: fix indent

File: cmd/manifest_tools.go [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+
+	"github.com/obolnetwork/charon/app/errors"
+	"github.com/obolnetwork/charon/cluster"
+	"github.com/obolnetwork/charon/cluster/manifest"
+	manifestpb "github.com/obolnetwork/charon/cluster/manifestpb/v1"
+)
+
+// loadClusterManifest returns the cluster manifest from the provided config. It returns true if
+// the cluster was loaded from a legacy lock file.
+// TODO(xenowits): Refactor to remove boolean in return values, ie, return only (cluster, error).
```
Comment (xenowits) [inline cmd/manifest_tools.go:20]: this is completed, you can rebase this PR

File: cmd/view_cluster_manifest.go [hunk 4]
```diff
@@ -0,0 +1,153 @@
...
+	"github.com/obolnetwork/charon/app/errors"
+)
+
+func newViewClusterManifestCmd(runFunc func(io.Writer, string) error) *cobra.Command {
+	var manifestFilePath string
+
+	cmd := &cobra.Command{
+		Use:   "view-cluster-manifest",
+		Short: "Shows cluster manifest contents",
+		Long:  `Opens and shows the specified cluster manifest by printing its content in JSON form.`,
```
Comment (xenowits) [inline cmd/view_cluster_manifest.go:27]:
```suggestion
		Long:  `Opens and shows the contents of the specified cluster manifest in a human-readable JSON format.`,
```

File: cmd/view_cluster_manifest.go [hunk 5]
```diff
@@ -0,0 +1,153 @@
...
+	cmd.Flags().StringVar(&manifestFilePath, "manifest-file", "cluster-manifest.pb", "The path to the cluster manifest file.")
+
+	return cmd
+}
+
+func runViewClusterManifest(out io.Writer, manifestFilePath string) error {
+	return viewClusterManifest(manifestFilePath, out)
+}
+
+func viewClusterManifest(manifestFilePath string, out io.Writer) error {
```
Comment (xenowits) [inline cmd/view_cluster_manifest.go:43]: nit: add godoc

File: cmd/view_cluster_manifest.go [hunk 6]
```diff
@@ -0,0 +1,153 @@
...
+		return err
+	}
+
+	clusterJSON, err := protoToJSON(cluster)
+	if err != nil {
+		return err
+	}
+
+	if _, err := fmt.Fprintln(out, string(clusterJSON)); err != nil {
+		return errors.Wrap(err, "cluster json output write")
```
Comment (xenowits) [inline cmd/view_cluster_manifest.go:55]: nit:
```suggestion
		return errors.Wrap(err, "write cluster json output")
```

File: cmd/view_cluster_manifest.go [hunk 7]
```diff
@@ -0,0 +1,153 @@
...
+			ret[formatKey(key)] = rawString(concrete)
+		}
+	}
+
+	return ret
+}
+
+// formatKey formats key from camel case to snake case.
+func formatKey(key string) string {
+	// remove "Json" suffix if any
```
Comment (xenowits) [inline cmd/view_cluster_manifest.go:105]:
```suggestion
	// Remove "Json" suffix if any
```

**Review by xenowits — APPROVED** [review]

File: cmd/addvalidators.go (hunk 1 above)
Comment (gsora) [reply cmd/addvalidators.go:110]: Yeah maybe we can move TestConfig in a separate file (it's in `addvalidators.go` now) and have `cmd` share it.

File: cmd/cmd.go (hunk 2 above)
Comment (gsora) [reply cmd/cmd.go:49]: wdym?

File: cmd/view_cluster_manifest.go (hunk 5 above)
Comment (gsora) [reply cmd/view_cluster_manifest.go]: I removed that function and replaced it with a plain `runViewClusterManifest` - I don't think this function needs godoc (isn't exported, meaning of the function is clear enough).

File: cmd/cmd.go (hunk 2 above)
Comment (xenowits) [reply cmd/cmd.go:49]: Nvm, when viewing the PR, i couldn't see a tab space

**Pull request merged**